### PR TITLE
Checkin mozlog source code to wpt repo

### PR DESCRIPTION
**Pull request by WeizhongX**
There are mozlog change required to support subsuite. Check in mozlog source code here to make it easier to debug/fix the issues in mozlog.

Once this is stable, we will upstream fixes made here, and revert this change.

### CODE REVIEW

**Comment by WeizhongX** [comment]
@jgraham can you pls take a look? thanks!

The failures on resources/tests seems unrelated.

**Comment by jgraham** [comment]
I just fixed https://bugzilla.mozilla.org/show_bug.cgi?id=1844321 upstream, so I suggest we wait until that reaches mozilla-central and update the vendored copy.

Also since we're expecting to make changes, I think we should try to get the tests running in our CI. That might be easier if we don't use `third_party` but some other directory like `third_party_modified`.

File: tools/third_party/mozlog/mozlog/handlers/valgrindhandler.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    Valgrind errors look like this:
+
+    ==60741== 40 (24 direct, 16 indirect) bytes in 1 blocks are definitely lost in loss
+                  record 2,746 of 5,235
+    ==60741==    at 0x4C26B43: calloc (vg_replace_malloc.c:593)
+    ==60741==    by 0x63AEF65: PR_Calloc (prmem.c:443)
+    ==60741==    by 0x69F236E: PORT_ZAlloc_Util (secport.c:117)
+    ==60741==    by 0x69F1336: SECITEM_AllocItem_Util (secitem.c:28)
+    ==60741==    by 0xA04280B: ffi_call_unix64 (in /builds/s1ave/m-in-l64-valgrind-000000000000/objdir/toolkit/library/libxul.so) # noqa
```
Comment (WeizhongX) [inline tools/third_party/mozlog/mozlog/handlers/valgrindhandler.py:34]: @jgraham I changed "slave" to "s1ave" here. Such word is not allowed in chromium. Do you have a suggestion what to do here?

Comment (jgraham) [reply tools/third_party/mozlog/mozlog/handlers/valgrindhandler.py:34]: I think you could replace it with `worker`; it's only example output so I don't think that will break anything, and actually matches what gecko CI uses these days.

**Comment by WeizhongX** [comment]
Use https://github.com/web-platform-tests/wpt/pull/41160 instead.

**Pull request closed without merging**